src/java.logging/share/classes/java/util/logging/FileHandler.java

```diff
@@ -511,6 +506,17 @@ private void openFiles() throws IOException {
                         channel = FileChannel.open(lockFilePath,
                                 CREATE_NEW, WRITE);
                         fileCreated = true;
+                    } catch (AccessDeniedException ade) {
+                        // Try again. If it doesn't work, then this will
+                        // eventually ensure that we increment "unique" and
+                        // use another file name.
```

Comment:
I believe this would be more clear if the comment was put inside the if statement, just before continue.
It might be good to add some words about what might be causing the AccessDeniedException as well... 
Maybe something like:

```
                    } catch (AccessDeniedException ade) {
                        // This can be either a temporary, or a more permanent issue.
                        // The lock file might be still pending deletion from a previous run
                        // (temporary), or the parent directory might not be accessible, etc...
                        // If we can write to the current directory, and this is a regular file, 
                        // let's try again.
                        if (Files.isRegularFile(lockFilePath, LinkOption.NOFOLLOW_LINKS)
                            && isParentWritable(lockFilePath)) {
                            // Try again. If it doesn't work, then this will
                            // eventually ensure that we increment "unique" and
                            // use another file name.
                            continue;
                        } else {
                            throw ade; // no need to retry
                        } ...
```

Reply:
I agree. I've updated the comments